app/logic/transcript.py

```diff
@@ -34,22 +32,20 @@ def getProgramTranscript(username):
     current user.
     """
     # Add up hours earned in a term for each program they've participated in
-    programData = (ProgramEvent
-        .select(Program, Event, fn.SUM(EventParticipant.hoursEarned).alias("hoursEarned"))
-        .join(Program)
-        .switch(ProgramEvent)
-        .join(Event)
+    
+    programData = (Event
+        .select(Event, fn.SUM(EventParticipant.hoursEarned).alias("hoursEarned"))
         .join(EventParticipant)
         .where(EventParticipant.user == username)
-        .group_by(Program, Event.term)
+        .group_by(Event.program, Event.term)
         .order_by(Event.term)
         .having(fn.SUM(EventParticipant.hoursEarned > 0)))
     transcriptData = {}
     for program in programData:
         if program.program in transcriptData:
-            transcriptData[program.program].append([program.event.term.description, program.hoursEarned])
+            transcriptData[program.program].append([program.term.description, program.hoursEarned])
         else:
-            transcriptData[program.program] = [[program.event.term.description, program.hoursEarned]]
+            transcriptData[program.program] = [[program.term.description, program.hoursEarned]]
```

Comment:
They were using the term `program` a lot of places here, but you are now selecting from events. This query should probably be selecting from program, not event